File: 05_Arrays/src/Reshape_the_Matrix_566.java
public class Reshape_the_Matrix_566 {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        if ((mat.length * mat[0].length) == (r*c)) {
            int[][] ansArr = new int[r][c];
            int row = 0;
            int col = 0;
            for (int i = 0; i < mat.length; i++) {
                for (int j = 0; j < mat[0].length; j++) {
                    ansArr[row][col] = mat[i][j];
                    col++;
                    if (col == c) {
                        col = 0;
                        row++;
                    }
                }
            }

            return ansArr;

        } else {
            return mat;
        }
    }

    public static void main(String[] args) {
    }
}

File: 05_Arrays/src/Reshape_the_Matrix_566.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    if ((mat.length * mat[0].length) === (r * c)) {
        let ansArr = [];
        let temp = [];

        for (let i = 0; i < mat.length; i++) {
            for (let j = 0; j < mat.length; j++) {
                temp.push(mat[i][j]);

                if(temp.length === c) {
                    ansArr.push(temp);
                    temp = [];
                    row = 0;
                }
            }
        }

        return ansArr;
    } else {
        return mat;
    }
};

console.log(matrixReshape([[1,2,3,4]],2,2))